TLSbo85240_fix_MXC_NAND_driver_issues_two_read_status_cmd.patch

Once bug 17963 is fixed, mxc_nand_dev_ready won't need to call
get_dev_status(). Once this is complete, get_dev_status() should only
get called after the status command has already been sent by
mxc_nand_command. Therefore, the send_cmd(NAND_CMD_STATUS) is removed
from  get_dev_status function.
===================================================================
diff -uNpr linux-2.6.19.2.orig/drivers/mtd/nand/mxc_nd.c linux-2.6.19.2/drivers/mtd/nand/mxc_nd.c
--- linux-2.6.19.2.orig/drivers/mtd/nand/mxc_nd.c	2007-04-25 17:28:14.000000000 -0600
+++ linux-2.6.19.2/drivers/mtd/nand/mxc_nd.c	2007-04-25 17:36:56.000000000 -0600
@@ -431,9 +431,6 @@ static u16 get_dev_status(void)
 	 */
 	NFC_BUF_ADDR = 1;
 
-	/* Send the Read status command before reading status */
-	send_cmd(NAND_CMD_STATUS, true);
-
 	/* Read status into main buffer */
 	NFC_CONFIG1 &= (~(NFC_SP_EN));
 	NFC_CONFIG2 = NFC_STATUS;
@@ -441,11 +438,11 @@ static u16 get_dev_status(void)
 	/* Wait for operation to complete */
 	wait_op_done(TROP_US_DELAY, 0, true);
 
+	/* Status is placed in first word of main buffer */
 	/* get status, then recovery area 1 data */
 	ret = mainBuf[0];
 	*((u32 *) mainBuf) = store;
 
-	/* Status is placed in first word of main buffer */
 	return ret;
 }
 
